src/peft/tuners/adalora/model.py

```diff
@@ -160,7 +162,11 @@ def _create_new_module(lora_config, adapter_name, target, **kwargs):
             from .bnb import SVDLinear4bit
 
         gptq_quantization_config = kwargs.get("gptq_quantization_config", None)
-        AutoGPTQQuantLinear = get_auto_gptq_quant_linear(gptq_quantization_config)
+
+        if is_gptqmodel_available():
+            QuantLinear = get_gptqmodel_quant_linear(gptq_quantization_config, device_map)
```

Comment:
Following from the previous comment about `device_map`: If gptqmodel absolutely needs this attribute, if it doesn't exist on the model, let's raise an error with a helpful error message for the user.

Reply:
@BenjaminBossan  Done. We need it to extract `device` so we can assign a device-accelerated/compatible kernel for it. There may be multiple `trainable` kernels for a `device` type.  If `none` is passed, we will assume `cpu` and pass a slow `cpu` only `torch` kernel so `none` works too without the device ctx.